src/pages/Context.mdx

```diff
@@ -21,20 +21,41 @@ const Example = React.memo(props => {
 });
 ```
 
-For example the code below will trigger rerender always as `context` object will be always newly created.
+> Examples and info is taken from the official [ReactJS](https://reactjs.org/docs/context.html#caveats) documentation.
+
+Context uses reference identity to determine when to re-render, there are some gotchas that could trigger
+unintentional renders in consumers when a provider’s parent re-renders.
+For example, the code below will re-render all consumers every time the Provider re-renders
+because a new object is always created for value:
 
 ```jsx
 const Example = props => {
   const { children } = props;
-  // ⛔ ({ value: 1 } === { value: 1 }) is false, so reference is different
-  const context = { value: 1 };
 
-  return <Provider value={context}>{children}</Provider>;
+  return ()
+    // ⛔ ({ value: 1 } === { value: 1 }) is false, so reference is different
+    <Provider value={{ value: 1 }}>
+      {children}
+    </Provider>
+  );
+};
+```
+
+To get around this, lift the value into the parent’s state:
+
+```tsx
+const Example = props => {
+  const { children } = props;
+  const [value] = useState(1);
+
+  return <Provider value={{ value: value }}>{children}</Provider>;
```

Comment:
I believe it's the same thing 🤔 

```
{ value: value } === { value: value } // false
```

If we will just do there this:

```suggestion
  return <Provider value={value}>{children}</Provider>;
```

There is no connection with previous example... So I propose to change there examples to be class components as it is in official docs, it will have more sense.

```tsx
class Example extends React.Component {
  state = { value: 1 }
 
  render() {
    const { children } = this.props
    
    // ⛔ ({ value: 1 } === { value: 1 }) is false, so reference is different
    return <Provider value={{ value: this.state.value }}>{children}</Provider>
  }
}
```

```tsx
class Example extends React.Component {
  state = { value: 1 }
 
  render() {
    const { children } = this.props

    return <Provider value={this.state}>{children}</Provider>
  }
}
```

Reply:
oops, you are right...

Reply:
copy/paste development is bad thing :)

Reply:
fixed